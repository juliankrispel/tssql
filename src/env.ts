import { ValueOf } from "./utils"

/**
 * Retrieve one env variable, throws if variable anything other than string
 */
export function env(name: string): string {
  const _var = process.env[name]
  if (typeof _var !== 'string') {
    throw new Error(`process.env['${name}'] must be type of string, currently '${typeof _var}'`)
  } 
  return _var
}
