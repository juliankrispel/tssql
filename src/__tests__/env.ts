import { env } from "../env"

describe('env', () => {
  it('returns env var', () => {
    expect(env('ENVIRONMENT')).toMatchSnapshot()
    expect(env('user')).toMatchSnapshot()
    expect(env('password')).toMatchSnapshot()
    expect(env('password')).toMatchSnapshot()
  })

  it('throws if env var is not defined or null', () => {
    expect(() => env('nullvar')).toThrowErrorMatchingSnapshot()
  })
})

