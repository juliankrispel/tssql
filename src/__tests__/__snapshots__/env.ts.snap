// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`env returns env var 1`] = `"test"`;

exports[`env returns env var 2`] = `"test-user"`;

exports[`env returns env var 3`] = `"test-password"`;

exports[`env returns env var 4`] = `"test-password"`;

exports[`env throws if env var is not defined or null 1`] = `"process.env['nullvar'] must be type of string, currently 'undefined'"`;
